src/main/java/org/alfresco/rest/api/impl/SiteMembershipRequestsImpl.java

```diff
@@ -479,5 +511,163 @@ public SiteInfo doWork() throws Exception
+	{
+    	personId = people.validatePerson(personId, true);
+
+		List<Invitation> invitations = getSiteInvitations(personId);
+		return createPagedSiteMembershipRequests(toSiteMembershipRequests(invitations), paging);
 	}
+
+    @Override
+    public CollectionWithPagingInfo<SiteMembershipRequest> getPagedSiteMembershipRequests(final Parameters parameters)
+    {
+        InvitationSearchCriteriaImpl criteria = new InvitationSearchCriteriaImpl();
+        criteria.setInvitationType(InvitationType.MODERATED);
+        criteria.setResourceType(ResourceType.WEB_SITE);
+
+        // Parse where clause properties.
+        Query q = parameters.getQuery();
+        if (q != null)
+        {
+            // Filtering via "where" clause.
+            MapBasedQueryWalker propertyWalker = createSiteMembershipRequestQueryWalker();
+            QueryHelper.walk(q, propertyWalker);
+
+            String siteId = propertyWalker.getProperty(PARAM_SITE_ID, WhereClauseParser.EQUALS, String.class);
+
+            if (siteId != null && !siteId.isEmpty())
+            {
+                criteria.setResourceName(siteId);
+            }
+
+            String personId = propertyWalker.getProperty(PARAM_PERSON_ID, WhereClauseParser.EQUALS, String.class);
+
+            if (personId != null && !personId.isEmpty())
+            {
+                criteria.setInvitee(personId);
+            }
+        }
+
+        List<Invitation> invitations = invitationService.searchInvitation(criteria);
+        return createPagedSiteMembershipRequests(toSiteMembershipRequests(invitations, true), parameters.getPaging());
+    }
+
+    @Override
+    public void approveSiteMembershipRequest(String siteId, String inviteeId, SiteMembershipApproval siteMembershipApproval)
+    {
+        SiteInfo siteInfo = sites.validateSite(siteId);
+        if (siteInfo == null)
+        {
+            throw new EntityNotFoundException(siteId);
+        }
+
+        // Set the site id to the short name (to deal with case sensitivity issues with
+        // using the siteId from the url)
+        siteId = siteInfo.getShortName();
+
+        // Validate invitation.
+        Invitation invitation = getSiteInvitation(inviteeId, siteId);
+        if (invitation == null || !(invitation instanceof ModeratedInvitation))
+        {
+            throw new RelationshipResourceNotFoundException(siteId, inviteeId);
+        }
+
+        ModeratedInvitation moderatedInvitation = (ModeratedInvitation) invitation;
+        ResourceType resourceType = moderatedInvitation.getResourceType();
+
+        if (!resourceType.equals(ResourceType.WEB_SITE) || !SiteVisibility.MODERATED.equals(siteInfo.getVisibility()))
+        {
+            // note: security, no indication that this has a different visibility
+            throw new RelationshipResourceNotFoundException(siteId, inviteeId);
+        }
+
+        invitationService.approve(invitation.getInviteId(), "");

```

Comment:
Wondering if a reason could be added as for the rejection.

Reply:
This option wasn't requested at this time, it can be added in the future.